modules/drivers/bigquery/test/metabase/driver/bigquery/query_processor_test.clj

```diff
@@ -59,6 +59,28 @@
                                         "FROM `v2_test_data.checkins` "
                                         "LIMIT 2")}
                  :type     :native
+                 :database (mt/id)})))))
+
+    (testing "queries with array result columns deserialize properly (issue #10275)"
+      (is (= [[["foo" "bar"]
+               [1 2]
+               [3.14159265359 0.5772156649]
+               [1234M 5678M]
+               [#t "2018-01-01T00:00Z[UTC]" #t "2018-12-31T00:00Z[UTC]"]
+               [#t "12:34" #t "20:01:13.230"]
+               [#t "1957-05-17T03:35Z[UTC]" #t "2018-06-01T01:15:34.120Z[UTC]"]
+               [#t "2014-09-27T20:30:00.450Z[UTC]" #t "2020-09-27T14:57:00.450Z[UTC]"]]]
+             (mt/rows
+              (qp/process-query
+               {:native   {:query (str "SELECT ['foo', 'bar'], "
```

Comment:
It would be nice to test for corner cases: empty array, nils in array.

Reply:
cool, adding an empty array. nils in arrays get `"Array cannot have a null element; error in writing field f8_"` as an error from BigQuery itself, and mixed types in an array get `Array elements of types {INT64, STRING} do not have a common supertype at [2:331]` so that's good.